### ISSUE TITLE
Adding EPG

### ISSUE THREAD

**Issue by 123mijnwebsite** [jwplayer/appletv#25]
Hello!,

I love it! And got everything to work :) Thank you for the good manual.

I know want to place a EPG (TV Guide) to a video (live stream). So when you choose a video and it shows you the description and button to play video I would like to show the EPG of that video/stream. Is this somehow easy to do with JW player? Have you got any tips or tricks how to get this done?

Thank you!

**Comment by RikHeijdens** [comment]
One way you could do this is by extending the ItemDetail template, you probably want to do something like the following:

- In `ItemDetail.js` when the view is registered make a XMLHttpRequest to your API that provides the EPG data.
- When the data has been loaded you can modify the document and insert your data using [TVML](https://developer.apple.com/library/prerelease/tvos/documentation/LanguagesUtilities/Conceptual/ATV_Template_Guide/index.html#//apple_ref/doc/uid/TP40015064-CH41-SW1) elements.

**Comment by 123mijnwebsite** [comment]
Thank you for you reply! Also This feed I've got is a .json format. In here there is a collection of all kinds of videos. First it shows the info about the program, time etc. But is it possible to extract the video urls in this feed and put them in a playlist? So all video's will be shown in 1 playlist.

So I got a .json file and want to show all videos in this .json file to show up in 1 playlist. 
I'm searching for it in Google but can't find anything to get this done.

Thanks again!

**Comment by josiekeller** [comment]
Yes! You'll need to parse the url, build a regular playlist with media item objects, and feed the media item objects into the player in order to play them. 

If you look at the source code under [appletv/jwplayer-appletv-web-app/js/PlaylistLoader.js](https://github.com/jwplayer/appletv/blob/master/jwplayer-appletv-web-app/js/PlaylistLoader.js) you can modify the xml playlist code to achieve this.

There is also more documentation on the player and playlist objects from Apple developers [here](https://developer.apple.com/library/tvos/documentation/TVMLJS/Reference/TVJSPlayer_Ref/index.html)

Hope this helps!.

**Comment by telergyhd** [comment]
123mijnwebsite... wij zijn ook bezig met het ontwikkelen van een TVOS app op basis van de JW Player maar kunnen nog wel wat extra handjes gebruiken... ook hebben we eventueel wel interesse in je source code daar wij hetzelfde bouwen denk ik...en app met live tv zenders en epg data

mijn email is s.kerstens@telergyhd.com hoop snel van je te horen

**Comment by johnluther** [comment]
@123mijnwebsite and @telergyhd , very exciting to hear that more people plan to use the code! Forks are great, but we also welcome pull requests to Master. :-)

JL

**Comment by 123mijnwebsite** [comment]
Still working on it... Can't figure out how to add a rss/json feed to show a tv guid (EPG) tips are welcome :)
